app/code/Magento/Bundle/Model/Product/Type.php

```diff
@@ -340,24 +341,22 @@ public function getWeight($product)
     {
         if ($product->getData('weight_type')) {
             return $product->getData('weight');
-        } else {
-            $weight = 0;
+        }
 
-            if ($product->hasCustomOptions()) {
-                $customOption = $product->getCustomOption('bundle_selection_ids');
-                $selectionIds = $this->serializer->unserialize($customOption->getValue());
-                $selections = $this->getSelectionsByIds($selectionIds, $product);
-                foreach ($selections->getItems() as $selection) {
-                    $qtyOption = $product->getCustomOption('selection_qty_' . $selection->getSelectionId());
-                    if ($qtyOption) {
-                        $weight += $selection->getWeight() * $qtyOption->getValue();
-                    } else {
-                        $weight += $selection->getWeight();
-                    }
+        if ($product->hasCustomOptions()) {
+            $customOption = $product->getCustomOption('bundle_selection_ids');
+            $selectionIds = $this->serializer->unserialize($customOption->getValue());
+            $selections = $this->getSelectionsByIds($selectionIds, $product);
+
+            return array_reduce($selections->getItems(), function ($weight, Selection $selection) use ($product) {
```

Comment:
Why we do not assign `0` to `$weight` anymore? Please provide a reproducible benchmark for such change.